use cases for volume snapshot (#24758)
kubernetes/website: content/en/docs/concepts/storage/volume-snapshots.md

`VolumeSnapshotClass` allows you to specify different attributes belonging to a `VolumeSnapshot`. These attributes may differ among snapshots taken from the same volume on the storage system and therefore cannot be expressed by using the same `StorageClass` of a `PersistentVolumeClaim`.

> Volume snapshots provide Kubernetes users with a standardized way to copy a volume's contents at a particular point in time without creating an entirely new volume. This functionality enables, for example, database administrators to backup databases before performing edit or delete modifications.

Reviewer: (nit):
```suggestion
Volume snapshots provide you with a standardized way to copy a volume's contents at a particular point in time, without creating an entirely new volume. This functionality enables, for example, database administrators to back up the storage for a database before performing edit or delete modifications.
```

> edit or delete modifications
You wouldn't usually snapshot the storage before running `UPDATE` or `DELETE`. `DROP TABLE`, you might. Is there wording that can highlight the second case as being a good fit?

If I have made a snapshot and now regret running that `DROP TABLE`, what do my next steps look like? How do I take the `VolumeSnapshot` and turn that into an outcome that the end users are happy with? (this could be something we leave for explaining in a separate GitHub issue).

* CSI drivers may or may not have implemented the volume snapshot functionality. The CSI drivers that have provided support for volume snapshot will likely use the csi-snapshotter. See [CSI Driver documentation](https://kubernetes-csi.github.io/docs/) for details.
* The CRDs and snapshot controller installations are the responsibility of the Kubernetes distribution.

> ### Use Cases for This Feature

Reviewer 2: I'm thinking even higher up in the intro. Maybe instead of diving head first into apis, the intro should start with some more conceptual information, like:

"The volume snapshot feature allows users to take snapshots of a PersistentVolumeClaim, as well as create a new PersistentVolumeClaim from a snapshot. Some of the use cases this enables include....

To use this feature, the following API resources are used..."

Author: So I moved this into the paragraph and nixed the separate section. I can move it all the way to the top and rewrite the first few sentences paragraph if you so desire.

Reviewer 2: Apologies for the delay. Yes, I think it would be great if the use case section could be moved to the very top

For more details, see
[Volume Snapshot and Restore Volume from Snapshot](/docs/concepts/storage/persistent-volumes/#volume-snapshot-and-restore-volume-from-snapshot-support).

> ## Use Cases for This Feature

Reviewer 2: I think it's nice to put the use cases at the top, before diving into the detailed api.

It may be good to understand how we distinguish this use case from volume cloning.

cc @xing-yang

Author: These are great points! I went ahead and moved the use cases into a subsection of the intro. I also added language to implicitly differentiate this from volume cloning.
